engine/access/rpc/backend/backend_accounts.go

```diff
@@ -80,6 +80,126 @@ func (b *backendAccounts) GetAccountAtBlockHeight(
 	return account, nil
 }
 
+// GetAccountBalance returns the account balance at the latest sealed block.
+// Alias for GetAccountBalanceAtLatestBlock
+func (b *backendAccounts) GetAccountBalance(ctx context.Context, address flow.Address) (uint64, error) {
+	return b.GetAccountBalanceAtLatestBlock(ctx, address)
+}
+
+// GetAccountBalanceAtLatestBlock returns the account balance at the latest sealed block.
+func (b *backendAccounts) GetAccountBalanceAtLatestBlock(ctx context.Context, address flow.Address) (uint64, error) {
+	sealed, err := b.state.Sealed().Head()
+	if err != nil {
+		err := irrecoverable.NewExceptionf("failed to lookup sealed header: %w", err)
+		irrecoverable.Throw(ctx, err)
+		return 0, err
+	}
+
+	sealedBlockID := sealed.ID()
+
+	if b.scriptExecutor != nil {
+		accountBalance, err := b.scriptExecutor.GetAccountBalance(ctx, address, sealed.Height)
+		if err != nil {
+			b.log.Debug().Err(err).Msgf("failed to get account balance at blockID: %v", sealedBlockID)
+			return 0, err
+		}
+
+		return accountBalance, nil
+	}
+
+	account, err := b.getAccountAtBlock(ctx, address, sealedBlockID, sealed.Height)
+	if err != nil {
+		b.log.Debug().Err(err).Msgf("failed to get account balance at blockID: %v", sealedBlockID)
+		return 0, err
+	}
+
+	return account.Balance, nil
+}
+
+// GetAccountBalanceAtBlockHeight returns the account balance at the given block height.
+func (b *backendAccounts) GetAccountBalanceAtBlockHeight(
+	ctx context.Context,
+	address flow.Address,
+	height uint64,
+) (uint64, error) {
+	if b.scriptExecutor != nil {
+		accountBalance, err := b.scriptExecutor.GetAccountBalance(ctx, address, height)
+
+		if err != nil {
+			b.log.Debug().Err(err).Msgf("failed to get account balance at height: %d", height)
+			return 0, err
+		}
+		return accountBalance, nil
+	}
+
+	account, err := b.GetAccountAtBlockHeight(ctx, address, height)
+	if err != nil {
+		return 0, err
+	}
+
+	return account.Balance, nil
+}
+
+// GetAccountKeys returns the account public keys at the latest sealed block.
+// Alias for GetAccountKeysAtLatestBlock
+func (b *backendAccounts) GetAccountKeys(ctx context.Context, address flow.Address) ([]flow.AccountPublicKey, error) {
+	return b.GetAccountKeysAtLatestBlock(ctx, address)
+}
+
+// GetAccountKeysAtLatestBlock returns the account public keys at the latest sealed block.
+func (b *backendAccounts) GetAccountKeysAtLatestBlock(ctx context.Context, address flow.Address) ([]flow.AccountPublicKey, error) {
+	sealed, err := b.state.Sealed().Head()
+	if err != nil {
+		err := irrecoverable.NewExceptionf("failed to lookup sealed header: %w", err)
+		irrecoverable.Throw(ctx, err)
+		return nil, err
+	}
+
+	sealedBlockID := sealed.ID()
+
+	if b.scriptExecutor != nil {
+		accountKeys, err := b.scriptExecutor.GetAccountKeys(ctx, address, sealed.Height)
```

Comment:
to support specifying a single key by index, you'll need to also add a `GetAccountKey()` method similar to `GetAccountKeys`.